control/tap-ctl-spawn.c

```diff
@@ -172,8 +174,42 @@ tap_ctl_get_child_id(int readfd)
 	return id;
 }
 
+/* Move the process to the nominated cgroup slice */
+static int
+tap_ctl_move_to_cgroup(int pid, const char *slice)
+{
+	FILE *f = NULL;
+	char *path;
+
+	if (!slice)
+		return -EINVAL;
+
+	path = malloc(FILENAME_MAX);
+	if (!path) {
+		EPRINTF("malloc failed: %d\n", errno);
+		return -errno;
+	}
+
+	snprintf(path, FILENAME_MAX, "/sys/fs/cgroup/cpu/%s/tasks", slice);
+	path[FILENAME_MAX-1] = '\0';
```

Comment:
I did say it was unlikely to happen, but this is taking input from the user, so there is nothing (that I can see anyway) to stop the user supplying something that causes this to overflow.